ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java

```diff
@@ -2950,21 +2956,33 @@ public Partition createPartition(Table tbl, Map<String, String> partSpec) throws
     int size = addPartitionDesc.getPartitionCount();
     List<org.apache.hadoop.hive.metastore.api.Partition> in =
         new ArrayList<org.apache.hadoop.hive.metastore.api.Partition>(size);
-    AcidUtils.TableSnapshot tableSnapshot = AcidUtils.getTableSnapshot(conf, tbl, true);
     long writeId;
     String validWriteIdList;
-    if (tableSnapshot != null && tableSnapshot.getWriteId() > 0) {
-      writeId = tableSnapshot.getWriteId();
-      validWriteIdList = tableSnapshot.getValidWriteIdList();
+
+    // In case of replication, get the writeId from the source and use valid write Id list
+    // for replication.
+    if (addPartitionDesc.getReplicationSpec() != null &&
+        addPartitionDesc.getReplicationSpec().isInReplicationScope() &&
+        addPartitionDesc.getPartition(0).getWriteId() > 0) {
+      writeId = addPartitionDesc.getPartition(0).getWriteId();
+      validWriteIdList =
```

Comment:
Ok. Underneath this code is using the valid write id list created using open transaction list of repl. So, this isn't wrong. But this may change subject to the changes because of other comment you have.

Reply:
Even that logic to create ValidWriteIdList based on all repl opened txns isn't right as it says that stats are valid for all these open txns but it isn't. Also, it sets high water mark based on 0th index in the replTxnsList map which might be pointing to wrong writeId compared to current txn's writeId. So, I doubt, this logic should be anyways removed.

Reply:
Done.